projects/fw_hw103/src/main.c

```diff
@@ -0,0 +1,63 @@
+#include <stdint.h>
+#include "adc.h"
+#include "delay.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
```

Comment:
I think you can remove some of these libraries such as soft_timer.h or delay.h since they are not being used.